packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1679,12 +1687,24 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                 return { ...attemptData, reason: "disconnected" };
             }
 
+            // If the GC version that this container is loaded from differs from the current GC version that this
+            // container is running, we need to regenerate the GC data and run full summary. This is used to handle
+            // scenarios where we upgrade the GC version because we cannot trust the data from the previous GC version.
+            let forceRegenerateData = false;
+            if (this.summaryGCVersion !== this.currentGCVersion && this.summaryGCVersion > 0) {
+                forceRegenerateData = true;
+                // We will regenerate the GC data by running full GC as per the current GC version. Update the version
+                // of GC data written in the summary to the current GC version.
+                this.summaryGCVersion = this.currentGCVersion;
```

Comment:
I don't think this should be here.. if the summary is nacked we have prematurely set this flag, right?

Reply:
I fixed this. This is now updated in `refreshLatestSummaryWithAck`. Can you please take a look.